src/coreclr/nativeaot/Runtime/loongarch64/AllocFast.S

```diff
@@ -0,0 +1,276 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <unixasmmacros.inc>
+#include "AsmOffsets.inc"
+
+// GC type flags
+GC_ALLOC_FINALIZE               = 1
+
+//
+// Rename fields of nested structs
+//
+OFFSETOF__Thread__m_alloc_context__alloc_ptr        = OFFSETOF__Thread__m_rgbAllocContextBuffer + OFFSETOF__gc_alloc_context__alloc_ptr
+OFFSETOF__Thread__m_alloc_context__alloc_limit      = OFFSETOF__Thread__m_rgbAllocContextBuffer + OFFSETOF__gc_alloc_context__alloc_limit
+
+
+
+// Allocate non-array, non-finalizable object. If the allocation doesn't fit into the current thread's
+// allocation context then automatically fallback to the slow allocation path.
+//  $a0 == MethodTable
+    LEAF_ENTRY RhpNewFast, _TEXT
+
+        // a1 = GetThread()
+#ifdef FEATURE_EMULATED_TLS
+        GETTHREAD_ETLS_1
+#else
+        INLINE_GETTHREAD  $a1
+#endif
+
+        //
+        // a0 contains MethodTable pointer
+        //
+        ld.w  $a2, $a0, OFFSETOF__MethodTable__m_uBaseSize
+
+        //
+        // a0: MethodTable pointer
+        // a1: Thread pointer
+        // a2: base size
+        //
+
+        // Load potential new object address into t3.
+        ld.d  $t3, $a1, OFFSETOF__Thread__m_alloc_context__alloc_ptr
+
+        // Determine whether the end of the object would lie outside of the current allocation context. If so,
+        // we abandon the attempt to allocate the object directly and fall back to the slow helper.
+        add.d  $a2, $a2, $t3
+        ld.d   $t4, $a1, OFFSETOF__Thread__m_alloc_context__alloc_limit
+        bltu  $t4, $a2, RhpNewFast_RarePath
+
+        // Update the alloc pointer to account for the allocation.
+        st.d  $a2, $a1, OFFSETOF__Thread__m_alloc_context__alloc_ptr
+
+        // Set the new objects MethodTable pointer
+        st.d  $a0, $t3, OFFSETOF__Object__m_pEEType
+
+        ori  $a0, $t3, 0
+        jirl  $r0, $ra, 0
+
+RhpNewFast_RarePath:
+        ori  $a1, $zero, 0
+        b  RhpNewObject
+    LEAF_END RhpNewFast, _TEXT
+
+// Allocate non-array object with finalizer.
+//  a0 == MethodTable
+    LEAF_ENTRY RhpNewFinalizable, _TEXT
+        ori  $a1, $zero, GC_ALLOC_FINALIZE
+        b  RhpNewObject
+    LEAF_END RhpNewFinalizable, _TEXT
+
+// Allocate non-array object.
+//  a0 == MethodTable
+//  a1 == alloc flags
+    NESTED_ENTRY RhpNewObject, _TEXT, NoHandler
+
+        PUSH_COOP_PINVOKE_FRAME $a3
+
+        // a3: transition frame
+
+        // Preserve the MethodTable in s0
+        ori  $s0, $a0, 0
+
+        addi.w  $a2, $zero, 0 // numElements
```

Comment:
Why does here use `addi.w` to clear $a2, instead of `ori`? I'm seeing `ori $zero, 0` more in other places.